### DISCUSSION TITLE
Clarification on overall work flow

### DISCUSSION THREAD

**Discussion by jeremysadkins** [Abe404/root_painter#138] [Q&A]
Hello,

It took me a minute to get this up and running, but I finally got the server and my local computer working together. I am now getting started and confused about the training data versus the project. From what I understand, I need to first create a training dataset with images that will not be part of the actual analysis. Then, after I use those files to create a model, I apply that model to the actual images I want segmentation for my study.

If that is correct, I would also like suggestions for the number of images to use in the trainer and time expectations. The first time I tried creating a training dataset, I used a little over 700 images (a mixture of tiffs and jpegs). I let it run all the data and barely processed half of it. Does this sound reasonable? I stopped the process and tried again with 50 images, which seems inadequate for training. I have 3/50, with an estimated 15 hours left to create the dataset.

To clarify my questions:
1. Do I need to create a training data set using images that will not be part of the final analysis to develop the model?
2. What is a reasonable number of images for the training dataset?
3. How long can I expect the creating dataset step to take?

**Comment by Abe404** [comment]
Hi Jeremy,

Apologies for the delayed response. Somehow I missed this message. 

> It took me a minute to get this up and running, but I finally got the server and my local computer working together. I am now getting > started and confused about the training data versus the project. From what I understand, I need to first create a training dataset with > images that will not be part of the actual analysis. Then, after I use those files to create a model, I apply that model to the actual > images I want segmentation for my study.

Yes, although the training images can be part of the actual analysis if you like, I just suggest creating a training dataset with smaller tiles if your images are large, for reasons discussed in the paper.

2. Depends on the variation in your dataset. 
3. Creating dataset should be quick. Training depends on how hard it is to train a model to segment your structure of interest and on the availability of your hardware. - You can get feedback from the metrics plot. 

Overall, I feel the mini-guide should be helpful for you here: https://github.com/Abe404/root_painter/blob/master/docs/mini_guide.md
